Description: CameraBin: unset GValues after finish using them
 GValue can contain allocated memory. Not unsetting it can cause memory
 leak. This patch adds g_value_unset() calls to various places that miss
 ones.
Author: Ratchanan Srirattanamet <ratchanan@ubports.com>
Forwarded: no
Last-Update: 2020-03-03
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/plugins/gstreamer/camerabin/camerabinsession.cpp
+++ b/src/plugins/gstreamer/camerabin/camerabinsession.cpp
@@ -1411,6 +1411,7 @@
         g_value_copy(oldRate, &rate);
         gst_structure_remove_all_fields(structure);
         gst_structure_set_value(structure, "framerate", &rate);
+        g_value_unset(&rate);
     }
 #if GST_CHECK_VERSION(1,0,0)
     caps = gst_caps_simplify(caps);
@@ -1532,6 +1533,8 @@
         gst_structure_remove_all_fields(structure);
         gst_structure_set_value(structure, "width", &w);
         gst_structure_set_value(structure, "height", &h);
+        g_value_unset(&w);
+        g_value_unset(&h);
     }
 
 #if GST_CHECK_VERSION(1,0,0)
